dags/dag_content/nextflow_tower_functions.py

```diff
@@ -20,3 +24,55 @@ def create_and_open_tower_workspace(tower_secret_key : str, platform: str, works
     tower_utils = TowerUtils(client_args)
     tower_utils.open_workspace(workspace_id=workspace_id)
     return tower_utils
+
+
+def get_latest_compute_environment(
+    tower_utils: TowerUtils,
+    stack_name: str,
+    workspace_id: str,
+) -> Dict[str, Optional[str]]:
+    """Gets the latest compute environments in the Nextflow Tower Workspace for a given stack
+    and workspace
+
+    Args:
+        tower_utils (TowerUtils): Authenticated TowerUtils object that contains tower client for use
+        stack_name (str): Name of the Tower stack
+        workspace_id (str): The ID number of the workspace you wish to retrieve the compute env for
+
+    Returns:
+        compute_env_ids (Dict): Latest SPOT and EC2 compute environments in a dictionary
+    """
+    compute_env_ids: dict[str, Optional[str]] = {"SPOT": None, "EC2": None}
+    endpoint = "/compute-envs"
+    params = {"workspaceId": workspace_id}
+    response = tower_utils.client.request("GET", endpoint, params=params)
+
+    # figure out highest compute env version by name
+    try:
+        ce_version = max(
+            [
+                int(comp_env["name"].split("-")[-1].replace("v", ""))
+                for comp_env in response["computeEnvs"]
+                if comp_env["platform"] == "aws-batch"
+                and comp_env["status"] == "AVAILABLE"
+            ]
+        )
+    except:
+        raise ValueError(
+            f"Cannot get compute environment version from Tower API request"
+        )
```

Comment:
It's a best practice to at least specify `except Exception:`. In this case, you might want to avoid losing the full stack trace by "extending" the exception using `from`. I tend to write the error message on a separate line to avoid additional indentation, but that's a [nit](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.). 

```suggestion
    except Exception as error:
        message = "Cannot get compute environment version from Tower API request"
        raise ValueError(message) from error
```